satpy/utils.py

```diff
@@ -363,6 +400,30 @@ def _get_sat_lonlat(data_arr, key_prefixes):
     return lon, lat
 
 
+def _get_satpos_from_platform_name(cth_dataset):
+    """Get satellite position if no orbital parameters in metadata.
+
+    Some cloud top height datasets lack orbital parameter information in
+    metadata.  Here, orbital parameters are calculated based on the platform
+    name and start time, via Two Line Element (TLE) information.
+
+    Needs pyorbital, skyfield, and astropy to be installed.
+    """
+    from pyorbital.orbital import tlefile
+    from skyfield.api import EarthSatellite, load
+    from skyfield.toposlib import wgs84
```

Comment:
Ah sorry, I guess this isn't `parallax` specifically, but there should probably be some extra defined for these extra dependencies.

Reply:
Skyfield has an optional dependency on astropy.  This optional dependency is triggered at least by:

```python
    height = wgs84.height_of(gc).to("km")
```

which is on line 423 in `utils.py`.  The `.to()` method leads to an import of `astropy.units`.  There may be other places with an astropy dependency; this is just the first place where tests fail if skyfield is present but astropy is not.

Since our functionality needs both skyfield and astropy, I will add both to `extras`.  As this functionality is indeed not limited to parallax correction, I will call it something else, perhaps `satpos_from_tle`?